src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
@fir3pho3nixx I do not have preference for one option or another (see previous comments on this PR) but I think using the `DependencyContext` is more robust because it gives the real dependencies, versus whatever dll's may be in the bin folder. Also loading just the dll's from the base directory may miss some assemblies. I am seeing this test failing [`Throws_helpful_exception_when_assembly_found_but_not_type`](https://github.com/castleproject/Windsor/blob/netcore/src/Castle.Windsor.Tests/TypeNameConverterTestCase.cs#L215)
which looks in Castle.Core for a type, but Castle.Core.dll is not present in the bin folder of the tests, even though it is loaded.